addons/test_mail/tests/test_mail_composer.py

```diff
@@ -430,6 +442,40 @@ def test_mail_composer_rights_portal(self):
             self.assertEqual(self.test_record.message_ids[0].body, '<p>Body text 2</p>')
             self.assertEqual(self.test_record.message_ids[0].author_id, portal_user.partner_id)
 
+    @users('user_rendering_restricted')
+    def test_mail_composer_rights_attachments(self):
+        """ Ensure a user without write access to a template can send an email"""
+        template_1 = self.template.copy({
+            'report_name': 'TestReport for {{ object.name }}.html',  # test cursor forces html
+            'report_template': self.test_report.id,
+        })
+        attachment_data = self._generate_attachments_data(2)
+        template_1.write({
+            'attachment_ids': [(0, 0, dict(a, res_model="mail.template", res_id=template_1.id)) for a in attachment_data]
+        })
+        with self.assertRaises(AccessError):
+            # ensure user_rendering_restricted has no write access
+            template_1.with_user(self.env.user).write({'name': 'New Name'})
+
+        template_1_attachments = template_1.attachment_ids
+        self.assertEqual(len(template_1_attachments), 2)
+
+        composer = self.env['mail.compose.message'].with_context(
+            self._get_web_context(self.test_record)
+        ).create({
+            'subject': 'Template Subject',
+            'body': '<p>Template Body</p>',
+            'template_id': template_1.id,
+            'attachment_ids': template_1_attachments.ids,
+            'partner_ids': [self.partner_employee_2.id],
+        })
+        composer._action_send_mail()
+
+        self.assertEqual(self.test_record.message_ids[0].subject, 'Template Subject')
+        self.assertEqual(
+            sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
```

Comment:
Shouldn't we have a report in attachments ?

Reply:
You should probably ask @mart-e about that, I just cherrypicked from #80233 and went no further than checking if the test failed without fix and passed with it.